File: src/bytetosize.py
import math
def bytes_to_size(bytes: int) -> str:
        sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', "PB", "EB", "ZB", "YB"]

        # Check for zero
        if bytes == 0:
            return 'n/a'

        # Get the index of the appropriate size
        i = int(math.floor(math.log(abs(bytes)) / math.log(1024)))

        # Check if the index is zero
        if i == 0:
            return f"{bytes} {sizes[i]}"

        # Convert the bytes to the appropriate size
        return f"{(bytes / (1024 ** i)):.1f} {sizes[i]}"

File: src/bytetosize.js
function bytesToSize(bytes) {
    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', "PB", "EB", "ZB", "YB"];

    if (bytes === 0) {
        return 'n/a';
    }

    const i = Math.floor(Math.log(Math.abs(bytes)) / Math.log(1024));

    if (i === 0) {
        return `${bytes} ${sizes[i]}`;
    }

    return `${(bytes / (1024 ** i)).toFixed(1)} ${sizes[i]}`;
}